src/features/canvass/components/LocationDialog/index.tsx

```diff
@@ -200,7 +204,18 @@ const LocationDialog: FC<LocationDialogProps> = ({
                   timestamp: new Date().toISOString(),
                 });
                 setShowSparkle(true);
-                goto('households');
+
+                // Generally, we want to back 2 steps, to the `households` step.
+                // But in case the user can end up here from another route in the future, we don't want to always
+                // back 2 steps, since that could back too far. So we have this check with a fallback behavior.
```

Comment:
I like this defensiveness, because I worry about this same thing. But ideally I'd like the history for this UI to behave more as a stack, i.e. where you push and pop rather than go "forward" and "back". Maybe the difference is very theoretical, and maybe it's not really possible, but that's how I think of the UI anyway.

Lets discuss at the code camp!